src/main/resources/graphql/icdc_queries.graphql

```diff
@@ -1288,5 +1491,295 @@ type QueryType {
     ] AS clinical_data_node_names
     RETURN clinical_data_node_names
     """)
-}
 
+    agentNodeData(study_code: String!): [AgentNodeData] @cypher(statement: """
+    MATCH (s:study)
+    WHERE s.clinical_study_designation = $study_code
+    OPTIONAL MATCH (sa:study_arm)<-[:of_study_arm]-(a:agent)
+    RETURN DISTINCT {
```

Comment:
Return statements (same for all the queries) probably can be simplified to 
`RETURN DISTINCT a {.*}`